slsDetectorSoftware/src/Module.cpp

```diff
@@ -1746,19 +1747,109 @@ void Module::setLEDEnable(bool enable) {
 // Pattern
 
 void Module::setPattern(const std::string &fname) {
-    uint64_t word;
-    uint64_t addr = 0;
-    FILE *fd = fopen(fname.c_str(), "r");
-    if (fd != nullptr) {
-        while (fread(&word, sizeof(word), 1, fd) != 0U) {
-            setPatternWord(addr, word); // TODO! (Erik) do we need to send
-                                        // pattern in 64bit chunks?
-            ++addr;
+    patternParameters pat;
+    memset(&pat, 0, sizeof(pat));
+    std::ifstream input_file;
+    input_file.open(fname.c_str(), std::ios_base::in);
+    if (!input_file.is_open()) {
+        throw RuntimeError("Could not open pattern file " + fname +
+                           " for reading");
+    }
+    std::string current_line;
+    while (input_file.good()) {
```

Comment:
Here it is also possible to loop directly on the reading
`for (std::string line; std::getline(input_file, line);){
...
}`

Reply:
done